src/agent/debuglet.ts

```diff
@@ -318,11 +369,34 @@ export class Debuglet extends EventEmitter {
         that.scheduleRegistration_(0 /* immediately */);
         that.emit('started');
       });
-
     });
   }
 
   /**
+   * isReady returns a promise that only resolved if the last breakpoint update
+   * happend within a duration (PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS). This
+   * feature is mainly used in Google Cloud Function (GCF), as it is a
+   * serverless environment and we wanted to make sure debug agent always
+   * captures the snapshots.
+   */
+  isReady(): Promise<void> {
+    if (Date.now() < this.breakpointFetchedTimestamp +
+            PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS) {
+      return Promise.resolve();
+    } else {
+      if (this.breakpointFetched.get() !== null) {
+        return this.breakpointFetched.get() as Promise<void>;
+      }
+      this.breakpointFetched.refresh();
+      (this.debuggeeRegistered.get() as Promise<void>).then(() => {
+        this.scheduleBreakpointFetch_(
+          0 /*immediately*/, true /*only fetch once*/);
```

Comment:
The `boolean` argument seems like a code smell. It suggests to me that we should refactor `scheduleBreakpointFetch` into two functions. One that does with the 'scheduling bits' and the other that deals with the 'fetching' bits. This usage here only needs to deal with the latter.

Reply:
there are two processes that will call `scheduleBreakpointFetch`, one is by the underlying scheduling from debuggee registration, the other is scheduled by `isReady()`. Here when the 'fetching' bits fails or has errors, it will also re call `scheduleBreakpointFetch` to reschedule. Therefore we cannot fully remove the Boolean variable to differentiate whether the scheduling is from debuggee or `isReady()`